---
title: "Class 12: Genome Informatics"
author: "Liana Melikian (A16675734)"
format: pdf
---

#Section 1. Proportion of G/G in a population
Downloaded a CSV files from Ensemble
Here we read this CSV file

Q5.
```{r}
mxl=read.csv("373531-SampleGenotypes-Homo_sapiens_Variation_Sample_rs8067378.csv")
head(mxl)
```

```{r}
table(mxl$Genotype..forward.strand.)
```

```{r}
table(mxl$Genotype..forward.strand.)/nrow(mxl)*100
```

Now let's look at a different population. I picked the TSI population.

```{r}
tsi=read.csv("373537-SampleGenotypes-Homo_sapiens_Variation_Sample_rs8067378.csv")
head(tsi)
```
```{r}
table(tsi$Genotype..forward.strand.)
```

```{r}
round(table(tsi$Genotype..forward.strand.)/nrow(tsi)*100,2)
```

This variant that is associated with childhood asthma is more frequent in the TSI population than the MXL population.

##Section 4: Population Scale Analysis

How many samples do we have?

```{r}
expr=read.table("rs8067378_ENSG00000172057.6.txt")
head(expr)
```

```{r}
nrow(expr)
```

Q13.
```{r}
counts=table(expr$geno)
counts
```
```{r}
medians=tapply(expr$exp, expr$geno,median)
medians
```


```{r}
library(ggplot2)
```

Q14.
```{r}
ggplot(expr)+aes(geno,exp,fill=geno)+
  geom_boxplot(notch=TRUE)
```
